plugins/importexport/crossref/filter/ArticleCrossrefXmlFilter.inc.php

```diff
@@ -238,36 +236,33 @@ function appendCollectionNodes($doc, $doiDataNode, $submission, $submissionFiles
 	 * Create and return component list node 'component_list'.
 	 * @param $doc DOMDocument
 	 * @param $submission PublishedArticle
-	 * @param $componentFiles array
+	 * @param $componentGalleys array
 	 * @return DOMElement
 	 */
-	function createComponentListNode($doc, $submission, $componentFiles) {
+	function createComponentListNode($doc, $submission, $componentGalleys) {
 		$deployment = $this->getDeployment();
 		$context = $deployment->getContext();
 		$request = Application::getRequest();
 
 		// Create the base node
 		$componentListNode =$doc->createElementNS($deployment->getNamespace(), 'component_list');
 		// Run through supp files and add component nodes.
-		foreach($componentFiles as $key => $componentFilesArray) {
-			// get galley best ID and locale
-			$keyParts = explode('-', $key);
-			foreach ($componentFilesArray as $componentFile) {
-				$componentNode = $doc->createElementNS($deployment->getNamespace(), 'component');
-				$componentNode->setAttribute('parent_relation', 'isPartOf');
-				/* Titles */
-				$componentFileTitle = $componentFile->getName($keyParts[1]);
-				if (!empty($componentFileTitle)) {
-					$titlesNode = $doc->createElementNS($deployment->getNamespace(), 'titles');
-					$titlesNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'title', $componentFileTitle));
-					$componentNode->appendChild($titlesNode);
-				}
-				// DOI data node
-				// TO-DO: bestId is missing for files
-				$resourceURL = $request->url($context->getPath(), 'article', 'download', array($submission->getBestArticleId(), $keyParts[0], $componentFile->getFileId()));
-				$componentNode->appendChild($this->createDOIDataNode($doc, $componentFile->getStoredPubId('doi'), $resourceURL));
-				$componentListNode->appendChild($componentNode);
+		foreach($componentGalleys as $componentGalley) {
+			$componentFile = $componentGalley->getFile();
+			$componentNode = $doc->createElementNS($deployment->getNamespace(), 'component');
+			$componentNode->setAttribute('parent_relation', 'isPartOf');
+			/* Titles */
+			$componentFileTitle = $componentFile->getName($componentGalley->getLocale());
+			if (!empty($componentFileTitle)) {
+				$titlesNode = $doc->createElementNS($deployment->getNamespace(), 'titles');
+				$titlesNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'title', $componentFileTitle));
+				$componentNode->appendChild($titlesNode);
 			}
+			// DOI data node
+			// TO-DO: bestId is missing for files
```

Comment:
Ah, yes, good catch! No, it is not relevant -- now that a galley has only one file and the galley URL (...article/view/articleId/galleyId)  is used... :-)